### PR TITLE
Nf 334 send create claim audit

### PR DESCRIPTION
**Pull request by rorticus0502**

### CODE REVIEW

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/models/create/CreateClaimAudit.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  claimReason: ClaimReason,
+  reclaimDutyPayments: Map[String, DutyPaid] = Map.empty,
+  bankDetails: BankDetails,
+  importerContactDetails: Option[ImporterContactDetails] = None,
+  repayTo: Option[RepayTo] = None,
+  entryDetails: EntryDetails,
+  itemNumbers: ItemNumbers,
+  uploads: Seq[UploadedFile] = Seq.empty,
+  fileTransferResults: Seq[FileTransferResult] = Seq.empty,
+  importerEori: Option[EoriNumber] = None
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/models/create/CreateClaimAudit.scala:39]: Is there a need for defaulting these optional and collection field values?  I don't think you are constructing this class anywhere and _not_ providing all of the values

Comment (rorticus0502) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/models/create/CreateClaimAudit.scala:39]: Good spot, thank you

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/ClaimService.scala [hunk 2]
```diff
@@ -63,4 +74,36 @@ class ClaimService @Inject() (connector: NIDACConnector) {
 
   }
 
+  def audit(success: Boolean, claim: Claim, claimResponse: CreateClaimResponse)(implicit
+    hc: HeaderCarrier,
+    ec: ExecutionContext
+  ): Unit = {
+
+    val audit = new CreateClaimAudit(
```
Comment (SteveSugden) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/ClaimService.scala:82]: Observation:  Do you need the "new" keyword for a case class?   Also, you could create an "apply" method on a companion `CreateClaimAudit` object that does this construction.   So the code here could then be reduced to 
``` val audit = CreateClaimAudit(success, claim, claimResponse)```

File: test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/base/TestData.scala [hunk 3]
```diff
@@ -147,4 +151,45 @@ trait TestData {
       uploadAnotherFile = Some(uploadAnotherFileAnswer)
     )
 
+  val claim: Claim = Claim(completeAnswers)
```
Comment (SteveSugden) [inline test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/base/TestData.scala:154]: You need to update from master - Claim now takes the user's EORI number

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/ClaimService.scala (hunk 2 above)
Comment (rorticus0502) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/ClaimService.scala:82]: will do

File: test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/base/TestData.scala (hunk 3 above)
Comment (rorticus0502) [reply test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/base/TestData.scala:154]: will do

**Review by SteveSugden — APPROVED** [review]
Looks good

**Pull request merged**